src/test/sql/sql_pipeline_test.cpp

```diff
@@ -399,18 +406,25 @@ TEST_F(SQLPipelineTest, GetResultTableNoOutput) {
 TEST_F(SQLPipelineTest, GetTimes) {
   auto sql_pipeline = SQLPipelineBuilder{_select_query_a}.create_pipeline();
 
-  EXPECT_THROW(sql_pipeline.translate_time_microseconds(), std::exception);
-  EXPECT_THROW(sql_pipeline.optimize_time_microseconds(), std::exception);
-  EXPECT_THROW(sql_pipeline.compile_time_microseconds(), std::exception);
-  EXPECT_THROW(sql_pipeline.execution_time_microseconds(), std::exception);
+  const auto& metrics = sql_pipeline.metrics();
+  ASSERT_EQ(metrics.statement_metrics.size(), 1u);
+  const auto& statement_metrics = metrics.statement_metrics[0].get();
+
+  const auto zero_duration = std::chrono::microseconds::zero();
+
+  EXPECT_EQ(statement_metrics.translate_time_micros, zero_duration);
+  EXPECT_EQ(statement_metrics.optimize_time_micros, zero_duration);
+  EXPECT_EQ(statement_metrics.compile_time_micros, zero_duration);
+  EXPECT_EQ(statement_metrics.execution_time_micros, zero_duration);
 
   // Run to get times
   sql_pipeline.get_result_table();
 
-  EXPECT_GT(sql_pipeline.translate_time_microseconds().count(), 0);
-  EXPECT_GT(sql_pipeline.optimize_time_microseconds().count(), 0);
-  EXPECT_GT(sql_pipeline.compile_time_microseconds().count(), 0);
-  EXPECT_GT(sql_pipeline.execution_time_microseconds().count(), 0);
+  EXPECT_GT(metrics.parse_time_micros, zero_duration);
```

Comment:
oh dear.... :cry:

Reply:
not quite sure I understand the comment...

Reply:
what happens if the parser finishes in less than a 1µs? or the tests run a system with a 4µs timer resolution (dunno if that exists) and parsing happens in less than that. Suddenly you got randomly failing tests, I suppose?

But since you just adjusted the tests, this is probably not the place for this discussion.

Reply:
well, i wrote the original tests so I'll probably be the one to fix them ;) any better idea how to do this? I'm aware that using clock time in tests is kinda bad, but i don't want to change lots of code just for the sake of testing it.

Reply:
In this case, if the requirement would be not to change the lib code, then no, I wouldn't have any. Probably wouldn't test it. But we want coverage and bohooo